### ISSUE TITLE
Lint/Debugger should not allow `pry`

### ISSUE THREAD

**Issue by ThHareau** [rubocop/rubocop#11658]
## Is your feature request related to a problem? Please describe.

Lint/Debugger is allowing `pry`. This is annoying because it should not go to production, and I would rather have it rejected, similary to a binding.pry

## Describe the solution you'd like

Rubocop is unhappy when `pry` is in the code

## Additional context

I think a solution would be to include the `pry` keyword in the config : https://github.com/rubocop/rubocop/blob/master/config/default.yml#LL1654-L1661. 

If you're ok with this I can do a PR. It seems also, according to the tests, that it was left out on purpose. Is there a specific reason for it?

**Comment by fatkodima** [comment]
From which library `pry` is coming from?

**Comment by koic** [comment]
> It seems also, according to the tests, that it was left out on purpose. Is there a specific reason for it?

@dvandersluis It looks like the test added in the following PR. Do you remember why `expect_no_offenses('pry')` was written?
https://github.com/rubocop/rubocop/pull/9504/files#diff-6b94f0f83e7ddc601d87c0822fd77f0b55746b8f73324b79f1e7f3a355f48bf8R186-R188

**Comment by dvandersluis** [comment]
The `pry` gem doesn’t add `pry` as a top level method as far as I’m aware. Is it coming from somewhere else?

I don’t remember specifically but the test was probably added to make sure methods defined in the config with receivers are not registered without them.

**Comment by koic** [comment]
Pry gem has `Object#pry` that starts a Pry REPL.
https://github.com/pry/pry/blob/v0.14.2/lib/pry/core_extensions.rb#L43-L49

It can be confirmed with the code below:

```ruby
require 'bundler/inline'

gemfile do
  source 'https://rubygems.org'

  gem 'pry'
end

forty_two = 42
pry
```

```console
% ruby example.rb
[1] pry(main)> forty_two
=> 42
```

**Comment by dvandersluis** [comment]
Fair enough! Generally people would use `binding.pry` there instead of just `pry`.

**Comment by ThHareau** [comment]
PR is here if someone wants to have a look :) https://github.com/rubocop/rubocop/pull/11669
